scripts/reliability_metrics.py

```diff
@@ -0,0 +1,194 @@
+#!/usr/bin/env python3.9
+import datetime
+import os
+import logging
+import threading
+import queue
+import re
+from ottopia_logging.logging_factory import LoggingFactory
+from ottopia_logging.log_level import LogLevel
+from ottopia_logging.logging_settings import LoggingSettings
+
+
+class LoggerSetting(LoggingSettings):
+    """
+    Logger settings
+    """
+
+    DEFAULT_COMPONENT_NAME: str = "keepalived-exporter"
+
+    DEFAULT_LOG_LEVEL: LogLevel = LogLevel.INFO
+
+    class Config:
+        """
+        Config for Logger settings
+        """
+
+        arbitrary_types_allowed = True
+
+
+
+class LineData:
+    def __init__(self, line: str, syslog_date: datetime.datetime):
+        self.line = line
+        self.syslog_date = syslog_date
+
+
+class Service:
+    def __init__(self, name: str, event: threading.Event):
+        self.name = name
+        self.event = event
+        self.queue = queue.Queue()
+        self.thread = threading.Thread(group=None, target=self.find)
+
+    def find(self):
+        fail_line: str = f'{self.name}.service: Main process exited, code=exited, status=1/FAILURE'
```

Comment:
maybe instead of parsing syslog entries we can query systemctl for the status of the service ?
like
```
$ systemctl is-failed sshd.service 
active

```

Reply:
https://unix.stackexchange.com/questions/110474/how-can-we-trace-problems-of-crashing-programs-in-linux